File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.py
class Solution(object):
    def findMaxAverage(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: float
        """
        maxSum = sum(nums[:k])
        curSum = maxSum

        for i in range(k, len(nums)):
            curSum += nums[i] - nums[i-k]
            print(curSum)
            maxSum = max(maxSum, curSum)

        return float(maxSum) / k

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.go
func findMaxAverage(nums []int, k int) float64 {
    ans := 0
    for i := 0; i < k; i++ {
        ans += nums[i]
    }

    cur := ans
    for i := k; i < len(nums); i++ {
        cur += nums[i] - nums[i-k]
        ans = max(ans, cur)
    }

    return float64(ans)/float64(k)
}
